### DISCUSSION TITLE
Error with Transfer Double Kernel Distance matrix

### DISCUSSION THREAD

**Discussion by Tom6161** [RangeShifter/RangeshiftR-tutorials#31] [Q&A] [answered]
Hey everyone,

I've tried building the dispersal module as follows:

`disp <-  Dispersal(Emigration = Emigration(EmigProb = 0.0), 
                   Transfer   = DispersalKernel(DoubleKernel = T, Distances = matrix(c(4.6, 68.6, 0.05), ncol = 3)),
                   Settlement = Settlement(StageDep = T, Settle = matrix(c(0,1,2,1,0,0), nrow = 3, ncol = 2)))`

and this runs fine, but when I run the code to join the parameters all together, it comes up with an error:

`s <- RSsim(simul = sim_test, land = land, demog = demo, dispersal = disp, init = init)`

`Error in validObject(s) : 
  invalid class “RSparams” object: Columns 1 and 2 of dispersal kernel traits (Distances) matrix must contain the two mean distance d1 and d2, with values >= 1000 (=landscape resolution (unless UseFullKernel=TRUE)) !`

Any ideas on how to fix please? My landscape module if it helps is as follows.

`carrycap <- c(0, 6, 11, 17, 23)`
`land <- ImportedLandscape(LandscapeFile = "Inputs/AlgarveMap.txt", 
                          Resolution = 1000, 
                          Nhabitats = 5, 
                          K_or_DensDep = carrycap, 
                          SpDistFile = "Inputs/seedling_corks.txt",
                          SpDistResolution = 1000)`

Also happy to provide any further info/script if it helps!

**Comment by annekathleenmalchow** [comment, marked as answer]
Dear Tom,
please read the error message carefully, as it already answers your question. It says the mean distances in the dispersal kernel (i.e. the first two entries, giving the mean dispersal distances for the double exponential kernel) must be larger than the landscape resolution, which is 1000 in your case (note that all distances are given in units of metres). This is also why the error only shows up when you combine the modules, because it concerns a restriction to the dispersal module that is posed by the landscape module. This restriction is needed to make sure that individuals can actually disperse into a different cell efficiently. Alternatively, you can use the option _UseFullKernel_ option; I quote from the manual:

> In order to separate emigration and transfer explicitly, and to avoid potential infinite re-sampling, the program requires the mean of the kernel to be greater or equal the cell resolution. This condition is relaxed only in the special case where emigration probability is set to be density-independent and the kernel is applied to the entire population without re-sampling (the ‘use full kernel’ option). Individuals which draw a short movement distance do not leave the natal cell/patch and implicitly become sedentary, and therefore the kernel itself defines the proportion of individuals which emigrate. When this option is selected, the emigration probability for those stages/sexes which disperse should be set to 1.0; otherwise, only a proportion of such individuals would use the kernel to determine whether or not they emigrate.

I hope this helps,
Anne

**Reply by Tom6161** [reply]
Thank you Anne, my bad!
